### ISSUE TITLE
Docs: Use `import.meta.dirname` in docs examples

### ISSUE THREAD

**Issue by kirkwaiblinger** [eslint/eslint#20737]
### Docs page(s)

https://eslint.org/docs/latest/use/configure/ignore#include-gitignore-files

### What documentation issue do you want to solve?

The ESLint docs currently use the old pattern `fileURLToPath(new URL(import.meta.url))` to access `__filename`/`__dirname`/`import.meta.filename`/`import.meta.dirname`. That's now unnecessary in all supported versions of node in eslint v10 (^22.13.0 || >=24), since `import.meta.filename`/`import.meta.dirname` are available.

I propose that all examples that use the old style be replaced with `import.meta.dirname`

### What do you think is the correct solution?

- Old style is yucky and verbose. New style is clear and concise.

### Participation

- [x] I am willing to submit a pull request for this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by DMartens** [comment]
I would agree that we should use `import.meta.dirname` but ESLint v10 still supports [^20.19.0](https://github.com/eslint/eslint/blob/e7e1889fca9b6044e08f41b38df20a1ce45808c8/package.json#L229) and the meta property was no longer experimental in [22.16.0](https://nodejs.org/docs/latest/api/esm.html#importmetadirname).
We will revisit this for the next major version of ESLint but as there is nothing to do, I am going to close this issue.

**Comment by kirkwaiblinger** [comment]
Oh, sorry! I brainfarted and went by the semver range from the rewrite repo: https://github.com/eslint/rewrite/blob/53df339563822e006302a3ce76d430575df1a436/package.json#L35
Thanks for catching that.

That said - I would still argue for using the `import.meta.dirname` in the docs. It's true it's not supported in all supported versions of node, but it is supported for most users (and [node 20 will be EOL by the end of April](https://nodejs.org/en/about/previous-releases), at which point it will be supported in all supported non-EOL versions of node). Furthermore, users who try to copy `import.meta.dirname` and encounter an error will have a very easy time googling and finding the fix, whereas it's not at all clear that the daunting `fileURLToPath(new URL(import.meta.URL));` should be replaced with `import.meta.filename` for users who have access to it.

Or, as an alternate proposal, would you be open to putting a comment `// Or, import.meta.dirname if available in your node version` wherever the `fileURLToPath` workaround appears? Something so that it's clear at a glance that all we're doing is accessing the data that is available on `import.meta` on recent versions of node.

**Comment by DMartens** [comment]
I do not think we will be updating the supported Node.js version even though version 20 will be out of date soon.
Personally I think a third option would be most appropriate: An info box in the the base documentation of the configuration detailing that you can use `import.meta.dirname` if possible.
I am against the first proposal of just using it: People may just want to copy and paste and it should work.
And for the second proposal it would be way too much noise in the configuration documentation.

Reopening the issue to see if other team members have a different opinion.

**Comment by mdjermanovic** [comment]
I think that providing multiple ways for resolving paths in different versions of Node.js would be too much noise in the documentation, even if that would be just a comment in the code or an info box. I think the current version that works in all supported versions of Node.js is fine for now, and that we should just replace it with the more concise version in the next major version when it becomes available in all supported versions of Node.js.

**Comment by lumirlumir** [comment]
How about marking this issue as blocked and moving it to the v11 triage board?

With Node.js 20 reaching end-of-life at the end of this month, it seems increasingly likely that v11 will include stable support for `import.meta.dirname` and `import.meta.filename`.

**Comment by mdjermanovic** [comment]
Sounds good to me, adding this to the v11 board.

**Comment by kirkwaiblinger** [comment]
Hi, sorry to push on an issue that already has an agreed-upon resolution and is blocked, but I have one other idea for something that is actionable in the meantime, if it's possible to consider.

What if we instead write it like so

```js
import path from "node:path";
import { fileURLToPath } from "node:url";

const dirname = fileURLToPath(import.meta.resolve('.'));
const gitignorePath = path.join(dirname, '.gitignore');

// etc
```

This would most clearly to signal to users with access to `import.meta.dirname` OR `__dirname` that they can replace the `const dirname` with their prefab version. Furthermore, when ESLint 11 is released, all that would change in the ESLint docs themselves is `const dirname = ...` &rarr; inlining `import.meta.dirname`, which would also make migration clearest for any users following the ESLint docs verbatim

---

There are of course many options for computing `dirname`, we could also do 
```js
const filename = fileURLToPath(import.meta.url);
const dirname = path.dirname(filename);
```
or whatever else seems best (though I might slightly prefer a succinct one-line version).

My only real criterion would be that `dirname` be computed and named, so that any following code is structurally identical to what would occur with `__dirname` or `import.meta.dirname`, rather than doing the relative path manipulations via file URLs.
